test/e2e/network_test.go

```diff
@@ -40,9 +61,374 @@ func testHybridOverlayRunning(t *testing.T) {
+	if err != nil {
+		return []v1.Event{}, err
+	}
+	var podEvents []v1.Event
+	for _, event := range eventList.Items {
+		if strings.Contains(event.InvolvedObject.Name, name) {
+			podEvents = append(podEvents, event)
+		}
+	}
+	return podEvents, nil
+}
+
+// createLinuxJob creates a job which will run the provided command with a ubi8 image
+func (tc *testContext) createLinuxJob(name string, command []string) (*batchv1.Job, error) {
+	linuxNodeSelector := map[string]string{"beta.kubernetes.io/os": "linux"}
+	return tc.createJob(name, ubi8Image, command, linuxNodeSelector, []v1.Toleration{})
+}
+
+//  createWinCurlerJob creates a Job to curl Windows server at given IP address
+func (tc *testContext) createWinCurlerJob(vm testVM, winServerIP string) (*batchv1.Job, error) {
+	winCurlerCommand := getWinCurlerCommand(winServerIP)
+	winCurlerJob, err := tc.createWindowsServerJob("win-curler-"+vm.GetCredentials().GetInstanceId(), winCurlerCommand)
+	return winCurlerJob, err
+}
+
+// getWinCurlerCommand generates a command to curl a Windows server from the given IP address
+func getWinCurlerCommand(winServerIP string) []string {
+	// This will continually try to read from the Windows Server. We have to try multiple times as the Windows container
+	// takes some time to finish initial network setup.
+	winCurlerCommand := []string{"powershell.exe", "-command", "for (($i =0), ($j = 0); $i -lt 10; $i++) { " +
+		"$response = Invoke-Webrequest -UseBasicParsing -Uri " + winServerIP +
+		"; $code = $response.StatusCode; echo \"GET returned code $code\";" +
+		"If ($code -eq 200) {exit 0}; Start-Sleep -s 10;}; exit 1"}
+	return winCurlerCommand
+}
+
+// createWindowsServerJob creates a job which will run the provided command with a Windows Server image
+func (tc *testContext) createWindowsServerJob(name string, command []string) (*batchv1.Job, error) {
+	windowsNodeSelector := map[string]string{"beta.kubernetes.io/os": "windows"}
+	windowsTolerations := []v1.Toleration{{Key: "os", Value: "Windows", Effect: v1.TaintEffectNoSchedule}}
+	return tc.createJob(name, windowsServerImage, command, windowsNodeSelector, windowsTolerations)
+}
+
+func (tc *testContext) createJob(name, image string, command []string, selector map[string]string,
+	tolerations []v1.Toleration) (*batchv1.Job, error) {
+	jobsClient := tc.kubeclient.BatchV1().Jobs(v1.NamespaceDefault)
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name + "-job",
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					Tolerations:   tolerations,
+					Containers: []v1.Container{
+						{
+							Name:            name,
+							Image:           image,
+							ImagePullPolicy: v1.PullIfNotPresent,
+							Command:         command,
+						},
+					},
+					NodeSelector: selector,
+				},
+			},
+		},
+	}
+
+	// Create job
+	job, err := jobsClient.Create(job)
+	if err != nil {
+		return nil, err
+	}
+	return job, err
```

Comment:
You can just return `jobsClient.Create(job)` here

Reply:
avoiding this as we are returning error